specifyweb/workbench/upload/treerecord.py

```diff
@@ -490,6 +490,8 @@ def _upload(self, to_upload: List[TreeDefItemWithParseResults], matched: Union[M
 
     def _do_insert(self, model, **kwargs):
         obj = model(**kwargs)
+        if hasattr(obj, 'fullname') and obj.fullname is None:
+            obj.fullname = obj.name
```

Comment:
Why was this change needed? can you mention the unit test this aimed to solve?

Reply:
by default, fullnames get reset at the very end anyways. If they don't, then you probably have a bug in changed_trees in upload.py

Reply:
Ya, I think there has to be a better place for this, but it solved an issue of the co->determination->taxon showing up correctly in the record results.  For some reason when the taxon.fullname was left null, then the issue occurred.  I'll look for a better solution.